envoy-control-source-consul/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/consul/services/ConsulServiceChanges.kt

```diff
@@ -222,3 +237,11 @@ class ConsulServiceChanges(
         }
     }
 }
+
+interface ServiceWatchPolicy {
+    fun shouldBeWatched(service: String, tags: List<String>): Boolean
+}
+
+object NoOpServiceWatchPolicy: ServiceWatchPolicy {
```

Comment:
There should be also "operational" implementation of ServiceWatchPolicy

Reply:
I will provide basic name and tags filtering policy.